pkg/pgmodel/ingestor/metric_batcher.go

```diff
@@ -72,7 +73,13 @@ func runMetricBatcher(conn pgxconn.PgxConn,
 		var err error
 		tableName, err = initializeMetricBatcher(conn, metricName, completeMetricCreationSignal, metricTableNames)
 		if err != nil {
-			firstReq.reportResult(fmt.Errorf("initializing the insert routine has failed with %w", err))
+			err := fmt.Errorf("initializing the insert routine for metric %v has failed with %w", metricName, err)
+			log.Error("msg", err)
+			firstReq.reportResult(err)
+		} else if tableName == "" {
+			err := fmt.Errorf("initializing the insert routine has returned an empty table name for metric name %v", metricName)
+			log.Error("msg", err)
+			firstReq.reportResult(err)
```

Comment:
Instead of doing it like this, why not return an error from `model.MetricTableName` in case table name is empty? That looks like a cleaner approach.

Reply:
I agree here.

Reply:
ok changed.